From bc9851aed0e856b7b6c0b8ceaa01e3faf9e9cf50 Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Tue, 1 Aug 2023 17:41:21 -0700
Subject: [PATCH 1/2] Add new sudoer group role

---
 plugins/sudoers/sudoers.in | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/plugins/sudoers/sudoers.in b/plugins/sudoers/sudoers.in
index 914d45e..2fae5b9 100644
--- a/plugins/sudoers/sudoers.in
+++ b/plugins/sudoers/sudoers.in
@@ -88,6 +88,10 @@ root ALL=(ALL:ALL) ALL
 ## Uncomment to allow members of group wheel to execute any command
 # %wheel ALL=(ALL:ALL) ALL
 
+%wheel ALL=(ALL) ALL
+%wheelnopw ALL=(ALL) NOPASSWD:ALL
+
+
 ## Same thing without a password
 # %wheel ALL=(ALL:ALL) NOPASSWD: ALL
 
-- 
2.41.0

