From 16d7e0fee0cf91eaac6c28c85db7c8b142101e3d Mon Sep 17 00:00:00 2001
From: Miguel Bernal Marin <miguel.bernal.marin@linux.intel.com>
Date: Thu, 23 Aug 2018 14:07:25 -0500
Subject: [PATCH 2/3] keep proxy settings environment variables

---
 plugins/sudoers/sudoers.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/plugins/sudoers/sudoers.in b/plugins/sudoers/sudoers.in
index bea3840..791218e 100644
--- a/plugins/sudoers/sudoers.in
+++ b/plugins/sudoers/sudoers.in
@@ -35,6 +35,9 @@
 ## You may wish to keep some of the following environment variables
 ## when running commands via sudo.
 ##
+## Proxy settings
+Defaults env_keep += "http_proxy https_proxy ftp_proxy no_proxy socks_proxy"
+##
 ## Locale settings
 # Defaults env_keep += "LANG LANGUAGE LINGUAS LC_* _XKB_CHARSET"
 ##
-- 
2.41.0

