From fe5b6849fc16c11432435d2d5142b06f68e4da58 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <michael.i.doherty@intel.com>
Date: Mon, 18 May 2015 16:52:05 +0100
Subject: [PATCH 3/4] visudo: Use sane default file

---
 plugins/sudoers/visudo.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/plugins/sudoers/visudo.c b/plugins/sudoers/visudo.c
index 533c952..eb9ae7f 100644
--- a/plugins/sudoers/visudo.c
+++ b/plugins/sudoers/visudo.c
@@ -132,6 +132,7 @@ main(int argc, char *argv[])
     int ch, oldlocale, editor_argc, exitcode = 0;
     bool quiet, strict, oldperms;
     debug_decl(main, SUDOERS_DEBUG_MAIN)
+    __attribute__ ((unused)) struct stat st = {0};
 
 #if defined(SUDO_DEVEL) && defined(__OpenBSD__)
     {
@@ -150,6 +151,10 @@ main(int argc, char *argv[])
     if (argc < 1)
 	usage(1);
 
+    if (stat("/etc/sudoers.d", &st) < 0) {
+        (void)mkdir("/etc/sudoers.d", 00755);
+    }
+
     /* Register fatal/fatalx callback. */
     sudo_fatal_callback_register(visudo_cleanup);
 
@@ -167,6 +172,9 @@ main(int argc, char *argv[])
     /* Parse sudoers plugin options, if any. */
     parse_sudoers_options();
 
+    /* Set default sudoers file, can be overriden with -f, --file */
+    sudoers_file = "/etc/sudoers.d/visudo";
+
     /*
      * Arg handling.
      */
-- 
2.18.0

