From c2e132743bead35cf5968616789ad51c9e2dcc7d Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Tue, 16 Jan 2024 12:37:32 -0800
Subject: [PATCH 3/3] visudo: Use sane default file

---
 plugins/sudoers/visudo.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/plugins/sudoers/visudo.c b/plugins/sudoers/visudo.c
index 735d7af..dad3b38 100644
--- a/plugins/sudoers/visudo.c
+++ b/plugins/sudoers/visudo.c
@@ -138,6 +138,7 @@ main(int argc, char *argv[])
     int ch, oldlocale, editor_argc, exitcode = 0;
     bool use_perms, use_owner, fflag;
     debug_decl(main, SUDOERS_DEBUG_MAIN);
+    __attribute__ ((unused)) struct stat st = {0};
 
 #if defined(SUDO_DEVEL) && defined(__OpenBSD__)
     {
@@ -156,6 +157,10 @@ main(int argc, char *argv[])
     if (argc < 1)
 	usage();
 
+    if (stat("/etc/sudoers.d", &st) < 0) {
+        (void)mkdir("/etc/sudoers.d", 00755);
+    }
+
     /* Register callbacks. */
     sudo_fatal_callback_register(visudo_cleanup);
     set_callbacks();
@@ -171,6 +176,9 @@ main(int argc, char *argv[])
     /* Parse sudoers plugin options, if any. */
     parse_sudoers_options(&ctx);
 
+    /* Set default sudoers file, can be overriden with -f, --file */
+    path_sudoers = "/etc/sudoers.d/visudo";
+
     /*
      * Arg handling.
      */
-- 
2.43.0

